src/Utils/inferrableType.ts

```diff
@@ -0,0 +1,40 @@
+import ts from "typescript";
+import { BaseType } from "../Type/BaseType";
+import { BooleanType } from "../Type/BooleanType";
+import { NullType } from "../Type/NullType";
+import { NumberType } from "../Type/NumberType";
+import { StringType } from "../Type/StringType";
+import { UndefinedType } from "../Type/UndefinedType";
+
+export function getInferrableType(expression: ts.Expression | undefined): BaseType | undefined {
+    // TODO: Ideally, this should also support function calls and property access expressions
```

Comment:
About function calls (a.k.a inferred response types), you'll need to use the typechecker to get the inferred type from a function. Probably like what I've done with kitajs:

https://github.com/arthurfiorette/kita/blob/48dc243d7e3f10363b33c135e758003de1912b83/packages/generator/src/util/node.ts#L41

The only problem is that sometimes functions return types are not manually typed, and when you do the above, it's generated ts.TypeNodes without a valid source. The good news is that the #1386 fixed it.

For property access expressions, I guess that it is also going to be similar to function response types.